### ISSUE TITLE
Having Cart and Checkout in the same page causes rate selection to malfunction 

### ISSUE THREAD

**Issue by senadir** [woocommerce/woocommerce-gutenberg-products-block#2073]
**Describe the bug**
When both Cart and Checkout are on the same page, selecting a rate on the second block will focus the first block and cause a jump, this is likely because of shipping rates using the same id, so an instanceId prefix or suffix should fix this.

**To Reproduce**

1. create a page with both Cart and Checkout
2. select a rate on the second block
3. it will jump and select the first block.

**Expected behavior**
you should stay on the same block.

**Comment by nerrad** [comment]
When we tackle this, I think we should prevent having both cart and checkout blocks inserted into the same page as opposed trying to fix them working together.  Shopper experience with them on the same page would be very poor and enabling the merchant to shoot themselves in the foot with that would be a bad idea in my opinion.

**Comment by mikejolley** [comment]
I don't think this is a poor experience tbh, and I know from the past some users do use this type of setup (which is definitely not optimal in woo core!) on things like one-page websites.

I just did some testing and the shipping options no longer cause a jump, no console errors, and everything works perfect. I think the changes we made to unify contexts and hooks has resolved this :) 

Marking resolved.
